project1/project1/application.py

```diff
@@ -59,3 +59,28 @@ def register():
 def details():
     data = User.query.all()
     return render_template("admin.html", data = data)
+
+# @app.route("/auth", methods = ["GET", "POST"])
+# def auth():
+#     request.method == "POST" 
+# 
+@app.route('/auth',methods=['POST', 'GET'])
+def auth():
+    if request.method == 'POST':
+        username=request.form.get('username')
+        password=request.form.get('password')
+        thisuser= db.query(User).get(username)
+        try:
+            if(username==thisuser.username) and (password==thisuser.password):
+                session['username'] = username
+                
+                return render_template("login.html", username = username)
+            else:
+                return render_template('registration.html',message="Please Enter Valid Password")
+        except:
+            return render_template('registration.html',message="Please Enter Valid Credentials")   
```

Comment:
Your work here missed the following

- render_template will change the UI but not the URL in browser. So you still see the /auth which is not accepted, rather redirect to the registration with proper message
- Handle the case when user bypass the login and tries to access /auth, in such case redict to /register with a message